src/EditorFeatures/Core/EditorConfigSettings/Data/Whitespace/WhitespaceSetting.cs

```diff
@@ -56,5 +69,79 @@ protected WhitespaceSetting(string description, OptionUpdater updater, SettingLo
             var location = new SettingLocation(isDefinedInEditorConfig ? LocationKind.EditorConfig : LocationKind.VisualStudio, fileName);
             return new WhitespaceSetting<TOption>(option, description, editorConfigOptions, visualStudioOptions, updater, location);
         }
+
+        private IEditorConfigStorageLocation2? GetEditorConfigStorageLocation()
+        {
+            return Key.Option.StorageLocations.OfType<IEditorConfigStorageLocation2>().FirstOrDefault();
+
+        }
+
+        public string? GetSettingName()
+        {
+            var storageLocation = GetEditorConfigStorageLocation();
+            return storageLocation?.KeyName;
+        }
+
+        public string GetDocumentation()
+        {
+            return Description;
+        }
+
+        public ImmutableArray<string>? GetSettingValues(OptionSet optionSet)
+        {
+            var storageLocation = GetEditorConfigStorageLocation();
+            var type = Key.Option.DefaultValue?.GetType();
+
+            if (storageLocation?.KeyName == "csharp_new_line_before_open_brace")
+            {
+                var strings = new List<string>();
+
+                _newLinesForBracesValues.Do(strings.Add);
+
+                return strings.ToImmutableArray();
+            }
+
+            if (storageLocation?.KeyName == "csharp_space_between_parentheses")
+            {
+                var strings = new List<string>();
+
+                _spaceWithinParenthesesValues.Do(strings.Add);
+
+                return strings.ToImmutableArray();
+            }
+
+            if (type == null)
+            {
+                return null;
+            }
+
+            if (type == typeof(bool))
```

Comment:
apologies if I already asked this and you answered it -lmk if that is the case.  Is it possible to move this to generic impls of whitespace settings?  E.g. the bool one only needs to know about bool options, etc.